### ISSUE TITLE
[Bug]: wrong IpcMainEvent.senderFrame.(osProcessId|processId), when handling an ipcRenderer.send from a preload of a cross-site nav

### ISSUE THREAD

**Issue by bughit** [electron/electron#33575]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

18.0.1 x64

### What operating system are you using?

Windows

### Operating System Version

win7 sp1 x64

### What arch are you using?

x64

### Expected Behavior

A cross-site nav causes a render replacement.

`IpcMainEvent.senderFrame.(osProcessId|processId)` in `ipcMain.on` from a preload `ipcRenderer.send` should be of the new renderer.

### Actual Behavior

`IpcMainEvent.senderFrame.(osProcessId|processId)` in `ipcMain.on` from a preload `ipcRenderer.send` is of the old renderer

### Testcase Gist URL

https://gist.github.com/bughit/8a366ff63eb5bb5885ebaad2cbbe9a77

**Comment by MarshallOfSound** [comment]
This is currently expected (yet undocumented) behavior that this property is _potentially_ invalid until the preload script has run.  Basically the underlying process handle in `RenderProcessHost` isn't swapped until `RenderProcessReady` is called which gets triggered _way_ further down the stack than our preload script logic.

The chromium `processId` properties are accurate, but the actual OS process ID isn't known until the process is deemed ready.

**Comment by bughit** [comment]
> the underlying process handle in RenderProcessHost isn't swapped until RenderProcessReady is called which gets triggered way further down the stack than our preload script

With "currently expected" are you, beyond explaining the current behavior, also saying it shouldn't be considered a bug? I'd argue that this swapping should happen before the preload so by the time any user code runs in the new renderer (which can invoke code in the browser), the browser is already aware the renderer has been replaced.

> The chromium processId properties are accurate

I'll recheck it, but that's not what I saw, both osProcessId and processId were of the old renderer process.

**Comment by MarshallOfSound** [comment]
> also saying it shouldn't be considered a bug?

Yes, the way that process swaps occur is tightly linked with the security model of Chromium (and in fact all this process swapping is deep in the `//content` layer).  The timing of swaps / when they become effective is very important and if we start messing with that we're undoubtedly going to cause bigger issues down the line.

Whether or not we call it a "bug" is irrelevant, because we won't be making changes to alter the behavior.  So at best you can call it a `wontfix` bug 😄 If you need a valid `osProcessId` you should wait for the `dom-ready` or `did-frame-finish-load` event at which point the process references by `webContents.mainFrame` or `event.senderFrame` will be accurate.

In theory though you shouldn't be depending on those process IDs for anything as they could in theory be swapped with a full sequence of navigation events (if for instance the renderer crashes and is restarted)
